src/bsp/platform/stm32f4/xi_bsp_mem_stm32f4.c

```diff
@@ -6,18 +6,36 @@
 
 #include <xi_bsp_mem.h>
 #include <stdlib.h>
+#include <stdio.h>
+#include <FreeRTOS.h>
+#include <portable.h>
 
 void* xi_bsp_mem_alloc( size_t byte_count )
 {
-    return ( void* )malloc( byte_count );
+    void* ptr = ( void* )pvPortMalloc( byte_count );
```

Comment:
Question here is should we use FreeRTOS's functions within the bsp or should we switch back to malloc and make FreeRTOS to use malloc and free as well by selecting heap3.c implementation.